client/src/main/java/org/camunda/bpm/client/task/ExternalTask.java

```diff
@@ -195,6 +196,30 @@
    * @return the business key
    */
   String getBusinessKey();
-
+  
+  /**
+   * Returns the value of the extension property for a given key or <code>null</code> if the property was not available.
+   * 
+   * @see ExternalTask#getExtensionProperties()
+   * @return the extension property, or <code>null</code> if not available
+   */
+  String getExtensionProperty(String propertyKey);
+
+  /**
+   * Returns all available extension properties. Extension properties must be
+   * defined at the external task activity inside the BPMN model and explicitly
+   * fetched (e.g. by calling
+   * {@link TopicSubscriptionBuilder#includeExtensionProperties(boolean)}) to be
+   * available.
+   * 
+   * If no extension properties are available the returned map will be empty.
+   * 
+   * @return a map of available extension properties which will be empty if no
+   *         property is available or extension properties were not fetched
+   *         (e.g. by calling
+   *         {@link TopicSubscriptionBuilder#includeExtensionProperties(boolean)})
```

Comment:
I am not that big a fan of duplicated information, the longer explanation says it all.
The "never null" part would be more interesting I guess.
```suggestion
   * @return a map of available extension properties, never <code>null</code>
```